### PR TITLE
Changes to fsl script and added Dockerfile 

### PR DESCRIPTION
**Pull request by lalet**

### CODE REVIEW

**Comment by lalet** [comment]
> +#Clone HCP pipelines
>  +RUN git clone https://github.com/Washington-University/Pipelines.git pipelines 
>  +RUN cp -r pipelines /usr/local/src/ 
>  +RUN rm -rf pipelines

While these lines were kept in a single statement separated with '\', it was throwing error. So kept them as three different lines.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Fsl-installer needs access to shell variable
+ENV SHELL /bin/bash
+
+# Install fsl to default dir
+RUN echo -e "/usr/local/src" | ./myFslInstallerScript.sh
+
+#Clone HCP pipelines
+RUN git clone https://github.com/Washington-University/Pipelines.git pipelines 
+RUN cp -r pipelines /usr/local/src/ 
+RUN rm -rf pipelines
```
Comment (glatard) [inline Dockerfile]: You could merge these 3 commands in a single RUN command, to save 2 layers.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ENV PATH ${FSLDIR}/bin:${PATH}
+ENV HCPPIPEDIR /usr/local/src/pipelines/PreFreeSurfer
+# FSL environment variables
+ENV FSLOUTPUTTYPE NIFTI_GZ
+ENV FSLTCLSH $FSLDIR/bin/fsltclsh
+ENV FSLWISH=$FSLDIR/bin/fslwish
+ENV FSLGECUDAQ=cuda.q
+# HCP environment variables
+ENV HCPPIPEDIR_PreFS=$HCPPIPEDIR/PreFreeSurferPipeline.sh
+ENV HCPPIPEDIR_Global=$HCPPIPEDIR/scripts
```
Comment (glatard) [inline Dockerfile]: Did you check that variable definitions have to be put in separate layers?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+RUN rm -rf pipelines
+
+# Set environment variables (run export not needed)
+ENV FSLDIR /usr/local/src/fsl
+ENV PATH ${FSLDIR}/bin:${PATH}
+ENV HCPPIPEDIR /usr/local/src/pipelines/PreFreeSurfer
+# FSL environment variables
+ENV FSLOUTPUTTYPE NIFTI_GZ
+ENV FSLTCLSH $FSLDIR/bin/fsltclsh
+ENV FSLWISH=$FSLDIR/bin/fslwish
```
Comment (glatard) [inline Dockerfile]: Use consistently '=' in variable assignments.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    bc \
+    git \
+    tar \
+    wget
+
+# Permit access to fsl installation script (must exist locally)
+ADD ./myFslInstallerScript.sh /myFslInstallerScript.sh
+
+# Fsl-installer needs access to shell variable
+ENV SHELL /bin/bash
```
Comment (glatard) [inline Dockerfile]: You mean your installer? Not sure...

**Review by glatard — CHANGES_REQUESTED** [review]
Looks good!

File: Dockerfile (hunk 2 above)
Comment (lalet) [reply Dockerfile]: Didn't understand what this means.

Comment (glatard) [reply Dockerfile]: Can't you group this all in the same command/layer?

Comment (lalet) [reply Dockerfile]: Ok let me try. So one group for FSL and one group for PreFreeSurfer.

File: Dockerfile [hunk 5]
```diff
@@ -13,26 +13,20 @@ RUN yum update -y && yum install -y \
 # Permit access to fsl installation script (must exist locally)
 ADD ./myFslInstallerScript.sh /myFslInstallerScript.sh
 
-# Fsl-installer needs access to shell variable
+
 ENV SHELL /bin/bash
```
Comment (glatard) [inline Dockerfile]: you removed the comment and not the code?

File: Dockerfile [hunk 6]
```diff
@@ -13,26 +13,20 @@ RUN yum update -y && yum install -y \
...
 #Clone HCP pipelines
-RUN git clone https://github.com/Washington-University/Pipelines.git pipelines 
-RUN cp -r pipelines /usr/local/src/ 
-RUN rm -rf pipelines
+RUN git clone https://github.com/Washington-University/Pipelines.git pipelines;cp -r pipelines /usr/local/src/;rm -rf pipelines;
 
 # Set environment variables (run export not needed)
-ENV FSLDIR /usr/local/src/fsl
-ENV PATH ${FSLDIR}/bin:${PATH}
-ENV HCPPIPEDIR /usr/local/src/pipelines/PreFreeSurfer
```
Comment (glatard) [inline Dockerfile]: why not all the ENV variables in the same ENV command?

File: Dockerfile [hunk 7]
```diff
@@ -13,26 +13,20 @@ RUN yum update -y && yum install -y \
...
+ENV FSLDIR=/usr/local/src/fsl
+ENV PATH=${FSLDIR}/bin:${PATH}
+ENV HCPPIPEDIR=/usr/local/src/pipelines/PreFreeSurfer
 # FSL environment variables
-ENV FSLOUTPUTTYPE NIFTI_GZ
-ENV FSLTCLSH $FSLDIR/bin/fsltclsh
-ENV FSLWISH=$FSLDIR/bin/fslwish
-ENV FSLGECUDAQ=cuda.q
+ENV FSLOUTPUTTYPE=NIFTI_GZ FSLTCLSH=$FSLDIR/bin/fsltclsh FSLWISH=$FSLDIR/bin/fslwish FSLGECUDAQ=cuda.q
```
Comment (glatard) [inline Dockerfile]: You could make it more readable using \

File: Dockerfile [hunk 8]
```diff
@@ -13,26 +13,20 @@ RUN yum update -y && yum install -y \
...
 ENV SHELL /bin/bash
 
 # Install fsl to default dir
 RUN echo -e "/usr/local/src" | ./myFslInstallerScript.sh
 
 #Clone HCP pipelines
-RUN git clone https://github.com/Washington-University/Pipelines.git pipelines 
-RUN cp -r pipelines /usr/local/src/ 
-RUN rm -rf pipelines
+RUN git clone https://github.com/Washington-University/Pipelines.git pipelines;cp -r pipelines /usr/local/src/;rm -rf pipelines;
```
Comment (glatard) [inline Dockerfile]: You could make it more readable using \ instead of ;

**Review by glatard — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 8 above)
Comment (lalet) [reply Dockerfile]: While keeping '\' this error gets thrown.
The command '/bin/sh -c git clone https://github.com/Washington-University/Pipelines.git pipelines     cp -r pipelines /usr/local/src/     rm -rf pipelines' returned a non-zero code: 129

File: Dockerfile (hunk 6 above)
Comment (lalet) [reply Dockerfile]: Yes I will change them.

File: Dockerfile (hunk 8 above)
Comment (lalet) [reply Dockerfile]: Ok I should keep the ';' as well. Working now.

Comment (glatard) [reply Dockerfile]: yep

**Pull request merged**